### ISSUE TITLE
Incorrect get_unwrap suggestion

### ISSUE THREAD

**Issue by rom1v** [rust-lang/rust-clippy#3508]
For this code:

```rust
fn main() {
    let v = vec![1, 2, 3];
    let _ = v.get(1).unwrap();
}
```

Clippy gives a correct suggestion:

```                                                                           
warning: called `.get().unwrap()` on a Vec. Using `[]` is more clear and more concise                                                                          
 --> src/main.rs:3:13                                                                                                                                          
  |                                                                                                                                                            
3 |     let _ = v.get(1).unwrap();                                                                                                                             
  |             ^^^^^^^^^^^^^^^^^ help: try this: `&v[1]`                                                                                                      
  |                                                                                                                                                            
  = note: #[warn(clippy::get_unwrap)] on by default                                                                                                            
  = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/master/index.html#get_unwrap
```

However, if the index is a variable:
```rust
fn main() {
    let v = vec![1, 2, 3];
    let index = 1;
    let _ = v.get(index).unwrap();
}
```

then the suggestion is wrong:

```
warning: called `.get().unwrap()` on a Vec. Using `[]` is more clear and more concise                                                                          
 --> src/main.rs:4:13                                                                                                                                          
  |                                                                                                                                                            
4 |     let _ = v.get(index).unwrap();                                                                                                                         
  |             ^^^^^^^^^^^^^^^^^^^^^ help: try this: `v[index]`                                                                                               
  |                                                                                                                                                            
  = note: #[warn(clippy::get_unwrap)] on by default                                                                                                            
  = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/master/index.html#get_unwrap
```

It should suggest `&v[index]` instead.

**Comment by oli-obk** [comment]
This has been fixed on master already. It will take some time to cycle into the stable toolchain
